src/components/homepage/header/search/Search.tsx

```diff
@@ -0,0 +1,162 @@
+import { punctuationRegex } from "../../../../regularExpressions/punctuationRegex";
+
+export function Search(): React.ReactElement {
+    const [searchInput, setSearchInput] = useState<string>("");
+    const [suggestedInput, setSuggestedInput] = useState<string>("");
+    const searchRef = useRef<HTMLDivElement>(null);
+    const navigate = useNavigate();
+    const hasSuggestions = suggestedInput?.length > 0;
+
+    (function handleOutsideClicks() {
+        useEffect(() => {
+            const detectOutsideClicks = (e: MouseEvent) => {
+                if (!searchRef.current?.contains(e.target as Node)) {
+                    setSuggestedInput("");
+                }
+            };
+
+            document.addEventListener("mousedown", detectOutsideClicks);
+            return () => {
+                document.removeEventListener("mousedown", detectOutsideClicks);
+            };
+        }, []);
+    })();
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>): void => {
+        if (hasSuggestions) {
+            navigate(`/pokemons/name/${suggestedInput}`);
+        } else {
+            const formData = new FormData(e.currentTarget);
+            const transmittedDataObject = Object.fromEntries(formData.entries());
+            const searchedPokemoName = transmittedDataObject.searchInput.toString().toLowerCase();
+            navigate(`/pokemons/name/${searchedPokemoName}`);
+        }
+    };
+
+    const handleFilterClick = () => {
+        navigate(`/filter/:gen`);
+    };
+
+    const handleChange = (e: React.ChangeEvent) => {
+        const clearedInput = (e.target as HTMLInputElement).value.replace(punctuationRegex, "");
+        setSearchInput(clearedInput);
+    };
+
+    return (
+        <Container>
+            <Form onSubmit={handleSubmit}>
+                <InputContainer ref={searchRef}>
+                    <Label $isShowingSuggestions={hasSuggestions}>
+                        <Input value={searchInput} onChange={handleChange} required />
+                        <SearchIconButton $isShowingSuggestions={hasSuggestions}>
+                            <ImgSvgContainer>
+                                <SearchIconImg href={SearchIcon} />
+                            </ImgSvgContainer>
+                        </SearchIconButton>
+                    </Label>
+                    <SearchSuggestions
+                        searchInput={searchInput.toLowerCase()}
+                        setSearchInput={setSearchInput}
+                        suggestedInput={suggestedInput}
+                        setSuggestedInput={setSuggestedInput}
+                    />
+                </InputContainer>
+                <ButtonsContainer>
+                    <SearchButton>Search</SearchButton>
+                    <FilterButton onClick={handleFilterClick}>Filter</FilterButton>
+                </ButtonsContainer>
+            </Form>
+        </Container>
+    );
+}
+
+const Container = styled(ContainerPrototype)``;
```

Comment:
What's the purpose of this ? you can just use the ContainerPrototype?

Reply:
It's more of a consistency thing, all my components start with container being an extension of container prototype,also makes adding styling later more straightforward